Formula/apache-arrow.rb

```diff
@@ -0,0 +1,38 @@
+class ApacheArrow < Formula
+  desc "Columnar in-memory analytics layer designed to accelerate big data"
+  homepage "https://arrow.apache.org/"
+  url "https://github.com/apache/arrow/archive/apache-arrow-0.4.1.tar.gz"
+  sha256 "a4fd4aaa3a2f671d031f0c31b4ed72df948d60c1b5aba84188b032da56e409e3"
+
+  head "https://github.com/apache/arrow.git"
+
+  # NOTE: remove ccache with Apache Arrow 0.5 and higher version
+  depends_on "ccache" => :recommended
+  depends_on "boost"
+  depends_on "cmake" => :build
```

Comment:
the dependency order here should be `cmake`, `boost`, `ccache`. So something like this:
```
# NOTE: remove ccache with Apache Arrow 0.5 and higher version
depends_on "cmake" => :build
depends_on "boost"
depends_on "ccache" => :recommended
```